pydantic/generics.py

```diff
@@ -140,6 +144,53 @@ def __concrete_name__(cls: Type[Any], params: Tuple[Type[Any], ...]) -> str:
         params_component = ', '.join(param_names)
         return f'{cls.__name__}[{params_component}]'
 
+    @classmethod
+    def __parameterized_bases__(cls, typevars_map: Mapping[TypeVarType, Type[Any]]) -> Iterator[Type[Any]]:
+        """Returns unbound bases of cls parameterised to given type variables
+
+
+        :param typevars_map: Dictionary of type applications for binding subclasses.
+            Given a generic class `Model` with 2 type variables [S, T]
+            and a concrete model `Model[str, int]`,
+            the value `{S: str, T: int}` would be passed to `typevars_map`.
+        :return: an iterator of generic sub classes, parameterised by `typevars_map`
+            and other assigned parameters of `cls`
+
+        e.g. for
+        ```
+        class A(GenericModel, Generic[T]):
+            ...
+
+        class B(A[V], Generic[V]):
+            ...
+
+        ```
+        then: `A[int] in B.__concrete_bases__({V: int})`
+        ```
+        """
+        if cls in _assigned_parameters:
+            # class is a partially "bound" form of a generic model
+            # We need to determine the mapping for the base_model parameters
+            mapped_types: Mapping[TypeVarType, Type[Any]] = {
+                key: typevars_map.get(value, value) for key, value in _assigned_parameters[cls].items()
+            }
+        else:
+            mapped_types = typevars_map
+        for base_model in cls.__bases__:
+            if issubclass(base_model, GenericModel):
+                if not hasattr(base_model, '__parameters__') or not base_model.__parameters__:
+                    # type is Generic or
+                    # base model is already concrete, and will be included transitively.
+                    continue
+                elif cls in _assigned_parameters and base_model in _assigned_parameters:
+                    # Subclass will be inferred via cls
+                    continue
+                else:
+                    base_parameters = tuple([mapped_types[param] for param in base_model.__parameters__])
+                    result = base_model.__class_getitem__(base_parameters)
```

Comment:
perhaps a comment to explain where we're at now in this `else` block (consistent with the ones above?)

Reply:
I've reworked it a bit and added some more explanation.  Could you check if it's any clearer now?